test/OrchardCore.Tests/Settings/SiteExtensionsTests.cs

```diff
@@ -0,0 +1,209 @@
+                    services.AddTransient<ISiteService>(_ => new SiteService(_site.Object));
+
+                    services.AddRouting();
+
+                    localizationStartup.ConfigureServices(services);
+                })
+                .Configure(app =>
+                {
+                    localizationStartup.Configure(app, new RouteBuilder(app), app.ApplicationServices);
+
+                    app.Run(context =>
+                    {
+                        var requestLocalizationOptions = context.RequestServices.GetService<IOptions<RequestLocalizationOptions>>().Value;
+
+                        Assert.Equal(expectedDefaultCulture, requestLocalizationOptions.DefaultRequestCulture.Culture.Name);
+                        Assert.Equal(expectedSupportedCultures, requestLocalizationOptions.SupportedCultures.Select(c => c.Name).ToArray());
+
+                        return Task.FromResult(0);
+                    });
+                });
+
+            using (var server = new TestServer(webHostBuilder))
+            {
+                var client = server.CreateClient();
+                var requestedCulture = "en";
+                var response = await client.GetAsync($"page/?culture={requestedCulture}&ui-culture={requestedCulture}");
+            }
+        }
+
+        private async Task RunTestWithAcceptLanguageHttpHeader(string acceptLanguages, string defaultCulture, string[] supportedCultures, string expectedDefaultCulture, string[] expectedSupportedCultures)
+        {
+            var localizationStartup = new OrchardCoreLocalization.Startup();
+            var webHostBuilder = new WebHostBuilder()
+                .ConfigureServices(services =>
+                {
+                    SetupSiteSettingsCultures(defaultCulture, supportedCultures);
+                    services.AddTransient<ISiteService>(_ => new SiteService(_site.Object));
+
+                    services.AddRouting();
+
+                    localizationStartup.ConfigureServices(services);
+                })
+                .Configure(app =>
+                {
+                    localizationStartup.Configure(app, new RouteBuilder(app), app.ApplicationServices);
+
+                    app.Run(context =>
+                    {
+                        var requestLocalizationOptions = context.RequestServices.GetService<IOptions<RequestLocalizationOptions>>().Value;
+
+                        Assert.Equal(expectedDefaultCulture, requestLocalizationOptions.DefaultRequestCulture.Culture.Name);
+                        Assert.Equal(expectedSupportedCultures, requestLocalizationOptions.SupportedCultures.Select(c => c.Name).ToArray());
+
+                        return Task.FromResult(0);
+                    });
+                });
+
+            using (var server = new TestServer(webHostBuilder))
+            {
+                var client = server.CreateClient();
+                client.DefaultRequestHeaders.AcceptLanguage.ParseAdd(acceptLanguages);
```

Comment:
Here there is only one culture added to the AcceptLanguage Request Header every time which is "it". Should you not add all the supported cultures except invariant culture since it doesn't support it?

Reply:
Yep, I did that to prove if the HTTP header contains unsupported culture it should work too, but if you need me add another test with another set of accept languages just let me know what set of languages that may we need to add to cover another set of cases

Reply:
Oh but this should always work because there's a fallback culture which is the default culture. So what we need to test is if the browser culture settings are set to a specific culture that is supported or unsupported what will be the result in that page in the html lang="" value. Maybe more a functional test in that case.